File: rest-service/src/main/java/mb/lib/model/ReportStatus.java
package mb.lib.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum ReportStatus
{
  NeverRun("never-run"),
  Queued("queued"),
  InProgress("in-progress"),
  Completed("completed"),
  Errored("errored"),
  Expired("expired");

  private final String externalValue;

  ReportStatus(String externalValue) {
    this.externalValue = externalValue;
  }

  @JsonValue
  public String getExternalValue() {
    return externalValue;
  }

  @JsonCreator
  public static ReportStatus fromExternalValue(String value) {
    for (var v : values())
      if (v.externalValue.equals(value))
        return v;

    throw new IllegalArgumentException(String.format(
      "Unrecognized enum value \"%s\" for enum %s",
      value,
      ReportStatus.class.getSimpleName()
    ));
  }
}

File: rest-service/src/main/kotlin/mb/lib/model/ReportStatus.kt
package mb.lib.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.node.TextNode

enum class ReportStatus(val externalValue: String) {
  NeverRun   ("never-run"),
  Queued     ("queued"),
  InProgress ("in-progress"),
  Completed  ("completed"),
  Errored    ("errored"),
  Expired    ("expired");

  @JsonValue
  fun toJSON(): JsonNode = TextNode(externalValue)

  companion object {
    @JsonCreator
    fun fromExternalValue(value: String): ReportStatus {
      for (v in values())
        if (v.externalValue == value)
          return v

      throw IllegalArgumentException("Unrecognized enum value \"$value\" for enum ${ReportStatus::class.simpleName}")
    }
  }
}
